src/iop/tonecurve.c

```diff
@@ -1293,19 +1306,35 @@ static gboolean dt_iop_tonecurve_motion_notify(GtkWidget *widget, GdkEventMotion
   GtkAllocation allocation;
   gtk_widget_get_allocation(widget, &allocation);
   int height = allocation.height - 2 * inset, width = allocation.width - 2 * inset;
-  c->mouse_x = CLAMP(event->x - inset, 0, width);
-  c->mouse_y = CLAMP(event->y - inset, 0, height);
+  int old_m_x = c->mouse_x;
+  int old_m_y = c->mouse_y;
+  c->mouse_x = event->x - inset;
+  c->mouse_y = event->y - inset;
+
+  const float mx = CLAMP(c->mouse_x, 0, width) / (float)width;
+  const float my = 1.0f - CLAMP(c->mouse_y, 0, height) / (float)height;
+
+  float multiplier;
+  GdkModifierType modifiers = gtk_accelerator_get_default_mod_mask();
+  if((event->state & modifiers) == GDK_CONTROL_MASK)
+  {
+    multiplier = dt_conf_get_float("darkroom/ui/scale_precise_step_multiplier");
+  }
+  else
+  {
+    multiplier = 1;
+  }
 
-  const float mx = c->mouse_x / (float)width;
-  const float my = 1.0f - c->mouse_y / (float)height;
+  const float dx = (old_m_x - c->mouse_x) * multiplier / (float)width;
```

Comment:
Something is rather broken with this math.
Without modifiers, when moving the point with mouse, it is clearly moving out-of-sync with the mouse.